checks/memory.rb

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+ 
+# Check for the presence of the DISABLE_INSTALLER_CHECKS
+# Environment Variable.  If it is nil, we dont want
+# to disable checking. Otherwise TRUE = nochecks
+# FALSE (or any other value) = check
+
+begin
+  require 'rubygems'
+rescue LoadError
+end
+require 'facter'
```

Comment:
You cannot use the facter gem directly, this is what we were talking about on IRC.  In Puppet 4, the gem isn't accessible to the installer.  You would need to shell out and call facter binary or use the original version you had.